NegativeLengthException.java

```diff
@@ -0,0 +1,22 @@
+public class NegativeLengthException extends RuntimeException {
+	private Throwable cause = null;

```

Comment:
Unused parameter. This is because you didn't `pull origin master` in your branch after you merged... Which in general will cause you to re-write code. Working correctly with git can save you a lot of work :)